### PR TITLE
SpotifyConnect supports scopes now,RecentTracks added

### PR DESCRIPTION
**Pull request by reol224**

### CODE REVIEW

File: backend/src/main/java/com/jos/spotifyclone/services/SpotifyConnect.java [hunk 1]
```diff
@@ -13,11 +13,15 @@
...
 import java.io.IOException;
 import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
 
 @Component
 public class SpotifyConnect {
     private final SpotifyApi spotifyApi;
-    private final AuthorizationCodeUriRequest authorizationCodeUriRequest;
+    //private final AuthorizationCodeUriRequest authorizationCodeUriRequest;
```
Comment (IVIURRAY) [inline backend/src/main/java/com/jos/spotifyclone/services/SpotifyConnect.java]: Is there any need to keep this commented line around?

File: backend/src/main/java/com/jos/spotifyclone/services/SpotifyConnect.java [hunk 2]
```diff
@@ -29,13 +33,30 @@ public SpotifyConnect(
                 .setClientSecret(secretId)
                 .setRedirectUri(SpotifyHttpManager.makeUri(redirectUri))
                 .build();
-        this.authorizationCodeUriRequest = spotifyApi.authorizationCodeUri().build();
+        //this.authorizationCodeUriRequest = spotifyApi.authorizationCodeUri().build();
```
Comment (IVIURRAY) [inline backend/src/main/java/com/jos/spotifyclone/services/SpotifyConnect.java]: Can we remove this if not needed?

File: backend/src/main/java/com/jos/spotifyclone/services/SpotifyConnect.java [hunk 3]
```diff
@@ -29,13 +33,30 @@ public SpotifyConnect(
...
+                "user-read-currently-playing " +
+                "user-read-playback-state " +
+                "user-modify-playback-state");
     }
 
 
     @PostConstruct
     public void openAuthWindow() {
-        final URI uri = authorizationCodeUriRequest.execute();
+        //final URI uri = authorizationCodeUriRequest.execute();
```
Comment (IVIURRAY) [inline backend/src/main/java/com/jos/spotifyclone/services/SpotifyConnect.java]: Same here...

File: backend/src/main/java/com/jos/spotifyclone/controller/RecentTracksController.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+@RestController
+public class RecentTracksController {
+
+    @Autowired
+    SpotifyConnect spotifyConnect;
+
+    @GetMapping("/recentTracks/")
+    public @ResponseBody
+    PagingCursorbased<PlayHistory> getRecentTracks() throws ParseException, SpotifyWebApiException, IOException {
+        //System.out.println((spotifyConnect.getSpotifyApi().getCurrentUsersRecentlyPlayedTracks().build().execute()));
```
Comment (IVIURRAY) [inline backend/src/main/java/com/jos/spotifyclone/controller/RecentTracksController.java]: Ideally it's good to not commit commented code to master. A classic code smell 😃 can you remove?

**Review by IVIURRAY — DISMISSED** [review]

File: backend/src/main/java/com/jos/spotifyclone/controller/RecentTracksController.java (hunk 4 above)
Comment (reol224) [reply backend/src/main/java/com/jos/spotifyclone/controller/RecentTracksController.java]: solved em all

**Review by IVIURRAY — APPROVED** [review]

**Pull request merged**